src/Network/Mattermost/WebSocket.hs

```diff
@@ -164,7 +164,7 @@ mmWithWebSocket (Session cd (Token tk)) recv body = do
         body (MMWS c health) `catch` propagate [mId, pId]
   WS.runClientWithStream stream
                       (T.unpack $ cdHostname cd)
-                      "/api/v4/websocket"
+                      (cdDirectory cd ++ "/api/v4/websocket")
```

Comment:
Same as above; in that case, I recommend factoring this out as a function to build the URL paths to avoid the duplication.